sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -233,15 +233,15 @@ def delete_entity(
         :raises: ~azure.core.exceptions.HttpResponseError
         """
 
-        if_match, if_not_match = _get_match_headers(kwargs=dict(kwargs, etag=kwargs.pop('etag', None),
+        if_match, _ = _get_match_headers(kwargs=dict(kwargs, etag=kwargs.pop('etag', None),
                                                                 match_condition=kwargs.pop('match_condition', None)),
                                                     etag_param='etag', match_param='match_condition')
         try:
             self._client.table.delete_entity(
                 table=self.table_name,
                 partition_key=partition_key,
                 row_key=row_key,
-                if_match=if_match or if_not_match or '*',
+                if_match=if_match or '*',
```

Comment:
just a question: why are you no longer checking `if_not_match`

Reply:
`if_not_match` is not supported by the service right now.